template/src/components/Toast.tsx

```diff
@@ -1,202 +1,180 @@
+      useNativeDriver: true,
+    }).start()
   }
 
-  render() {
-    const {message, type} = this.state
-    return (
-      <Animated.View
+  useEffect(() => {
+    Emitter.on(TOAST_EVENTS.showToastMessage, displayMessage)
+    return () => {
+      Emitter.rm(TOAST_EVENTS.showToastMessage)
+    }
+  }, [])
+
+  return (
+    <Animated.View
+      style={[
+        styles.container,
+        {
+          transform: [{translateY: offset}],
+          opacity: opacity,
+          paddingBottom: insets.bottom,
+        },
+      ]}>
+      <View
         style={[
-          styles.container,
+          styles.messageContainer,
           {
-            transform: [{translateY: this.offset}],
-            opacity: this.opacity,
-            backgroundColor: this.messageColor(),
+            backgroundColor: messageContent[state.type].color,
           },
         ]}>
-        <TouchableOpacity
-          activeOpacity={1}
-          style={styles.messageContainer(this.messageColor())}
-          onPress={() => {
-            this.dismiss()
-          }}>
-          <Text style={styles.textStyle}>
-            {message}
-          </Text>
+        <Image source={Images.info} style={styles.icon} />
+        <View style={styles.textContent}>
+          <Text style={styles.titleStyle}>{state.message}</Text>
+          {!!state.subMessage && <Text style={styles.textStyle}>{state.subMessage}</Text>}
+        </View>
+        <TouchableOpacity onPress={dismiss} hitSlop={hitSlop}>
+          <Image source={Images.close} style={styles.icon} />
         </TouchableOpacity>
-      </Animated.View>
-    )
-  }
+      </View>
+    </Animated.View>
+  )
+}
+
+export const showToast = (args: IToastState) => {
+  Emitter.emit(TOAST_EVENTS.showToastMessage, args)
 }
 
-const styles = StyleSheet.create<IStyleSheet | any>({
+const styles = StyleSheet.create({
   container: {
-    height: HEIGHT,
     zIndex: 9999,
     position: 'absolute',
-    top: 0,
+    bottom: 0,
     left: 0,
     right: 0,
+    paddingHorizontal: metrics.paddingHorizontal,
   },
-  messageContainer: (backgroundColor: string): ViewStyle => ({
+  messageContainer: {
     flex: 1,
-    justifyContent: 'flex-end',
-    alignItems: 'center',
-    backgroundColor,
-  }),
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'flex-start',
+    borderRadius: metrics.borderRadius,
+    paddingHorizontal: metrics.xs,
+    paddingVertical: metrics.xxs,
+  },
+  textContent: {
+    flex: 1,
+    paddingHorizontal: metrics.xxs,
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    justifyContent: 'center',
+  },
+  titleStyle: {
+    fontSize: FontSizes.title,
+    color: colors.white,
+  },
   textStyle: {
-    fontSize: FontSizes.span,
-    marginBottom: metrics.marginVertical,
-    textAlign: 'center',
-    colors: colors.white,
+    fontSize: FontSizes.body,
+    color: colors.white,
+  },
+  icon: {
+    width: metrics.icon,
+    aspectRatio: 1,
```

Comment:
should we add height equal with width in this style?

Reply:
using `aspectRatio: 1` is the same with that